src/api.ts

```
@@ -1,10 +1,11 @@
 import { MovieData } from './types/MovieData';
 import { ResponseError } from './types/ReponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_KEY = import.meta.env.VITE_OMDB_API_KEY;
+const API_URL = `https://www.omdbapi.com/?apikey=${API_KEY}`;
```

Storing the API_URL using the possibly-undefined API_KEY at module top makes it harder to handle a missing key gracefully. Consider constructing the full URL inside getMovie after validating API_KEY so you can return a clear error before performing fetch.